src/ddscxx/src/org/eclipse/cyclonedds/pub/AnyDataWriterDelegate.cpp

```diff
@@ -22,10 +22,15 @@
 #include <org/eclipse/cyclonedds/core/ScopedLock.hpp>
 #include <org/eclipse/cyclonedds/topic/BuiltinTopicCopy.hpp>
 #include <dds/dds.h>
-#include <dds/ddsc/dds_data_allocator.h>
+#include <dds/ddsc/dds_loan_api.h>
 
 #include "dds/ddsi/ddsi_sertopic.h"
 #include "dds/ddsi/q_protocol.h"
+#include "dds/features.hpp"
+
+#ifdef DDSCXX_HAS_SHM
+#include <dds/ddsi/shm_transport.h>
```

Comment:
Not for this PR but we probably need to expose the functionality in a C header and be able to use it without requiring the `#ifdef` here ...